eternalcore/src/main/java/com/eternalcode/core/EternalCore.java

```diff
@@ -488,24 +493,23 @@ public void onDisable() {
     }
 
     private void softwareCheck() {
-        try {
-            Class.forName("org.spigotmc.SpigotConfig");
-            this.isSpigot = true;
-        } catch (ClassNotFoundException exception) {
-            this.getLogger().warning("Your server running on unsupported software, use spigot/paper minecraft software and other spigot/paper 1.17x forks");
-            this.getLogger().warning("We recommend using paper, download paper from https://papermc.io/downloads");
-            this.getLogger().warning("WARRING: Supported minecraft version is 1.17-1.18x");
+        Logger logger = this.getLogger();
+        Environment environment = PaperLib.getEnvironment();
+
+        if (!environment.isSpigot()) {
+            logger.warning("Your server running on unsupported software, use spigot/paper minecraft software and other spigot/paper 1.19x forks");
+            logger.warning("We recommend using paper, download paper from https://papermc.io/downloads");
+            logger.warning("WARRING: Supported minecraft version is 1.17-1.19x");
+            return;
         }
 
-        if (this.isSpigot) {
-            this.getLogger().info("Your server running on supported software, congratulations!");
-            this.getLogger().info("Server version: " + this.getServer().getVersion());
+        if (environment.isVersion(17)) {
```

Comment:
To chyba nie powinno tak działać
Sprawdzasz czy wersja == 1.17

Tutaj log z konsoli 
```console
[22:42:50 WARN]: [EternalCore] EternalCore no longer supports your version, be aware that there may be bugs!
```